### PR TITLE
refactor(core): change pass_local_clients internals to Command builder

### LINKED ISSUE #1648: Change pass_local_client to use new Command builder under the hood
`pass_local_client` should use command builder internally.

### PR DESCRIPTION
**Pull request by Panaetius**
Adds Command builder inside pass_local_client

Closes #1648

### CODE REVIEW

File: renku/core/commands/client.py [hunk 1]
```diff
@@ -66,41 +59,30 @@ def pass_local_client(
...
-            client = ctx.ensure_object(LocalClient)
+        cmd = Command().command(method)
 
-        stack = contextlib.ExitStack()
+        if not ignore_std_streams:
+            cmd = cmd.track_std_streams()
 
-        # Handle --isolation option:
-        if get_git_isolation():
-            client = stack.enter_context(client.worktree())
```
Comment (mohammad-sdsc) [inline renku/core/commands/client.py:80]: What happens to isolation then? If we don't want to support, we should remove the command line option and update the docs.

**Review by mohammad-sdsc — COMMENTED** [review]
LGTM. I've made a comment about isolation support.

File: renku/core/commands/client.py (hunk 1 above)
Comment (Panaetius) [reply renku/core/commands/client.py:80]: It was in the `Command` class, but I move it around to make it more explicit.

Honestly, `--isolation` is one the features in renku that I don't fully understand (how it works/when it is useful). So I wouldn't feel comfortable making a decision to drop it or not. We could open an issue to investigate.

**Review by mohammad-sdsc — APPROVED** [review]
Thanks!

**Pull request merged**